packages/js/passkeys-next-auth-provider/src/index.ts

```diff
@@ -0,0 +1,71 @@
+import { Tenant } from "@teamhanko/passkeys-sdk";
+import { JWTPayload, JWTVerifyResult, createRemoteJWKSet, jwtVerify } from "jose";
+import CredentialsProvider from "next-auth/providers/credentials";
+
+export const DEFAULT_PROVIDER_ID = "passkeys";
+
+export function PasskeyProvider({
+	tenant,
+	authorize: authorize,
+	id = DEFAULT_PROVIDER_ID,
+}: {
+	tenant: Tenant;
+	/**
+	 * Called after the JWT has been verified. The passed-in `userId` is the value of the `sub` claim of the JWT.
+	 *
+	 * The `userId` can safely be used to log the user in, e.g.:
+	 *
+	 * @example
+	 * async function authorize({ userId }) {
+	 *     const user = await db.users.find(userId);
+	 *
+	 *     if (!user) return null;
+	 *
+	 *     return user;
+	 * }
+	 */
+	authorize?: (data: { userId: string; token: JWTPayload }) => any;
+	id?: string;
+}) {
+	const url = new URL(`${tenant.config.tenantId}/.well-known/jwks.json`, tenant.config.baseUrl);
+	const JWKS = createRemoteJWKSet(url);
+
+	return CredentialsProvider({
+		id,
+		credentials: {
+			/**
+			 * Token returned by `passkeyApi.login.finalize()`
+			 */
+			finalizeJWT: {
+				label: "JWT returned by /login/finalize",
+				type: "text",
+			},
+		},
+		async authorize(credentials) {
+			const jwt = credentials?.finalizeJWT;
+			if (!jwt) throw new Error("No JWT provided");
+
+			let token: JWTVerifyResult;
+			try {
+				token = await jwtVerify(jwt, JWKS);
+			} catch (e) {
+				console.warn("JWT verification failed", e);
+				return null;
+			}
+
+			const userId = token.payload.sub;
+			if (!userId) {
+				console.error('JWT does not contain a "sub" claim');
+				return null;
+			}
```

Comment:
It should also be checked when the JWT was created, because currently the JWT does not include the `exp` (expiration time) property. So every created JWT would be considered valid, that should not be the case.

A JWT should not be considered valid when it is older than a few seconds (e.g. 5 seconds). But maybe it is better to add the `exp` property to the JWT.

Reply:
Added a 60s exp check for now, have not tested it yet though